src/encoded/commands/purge_item_type.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import argparse
+import logging
+import structlog
+import transaction
+from pyramid.paster import get_app
+
+from dcicutils.env_utils import is_stg_or_prd_env
+from snovault import STORAGE
+from snovault.elasticsearch.indexer_utils import get_uuids_for_types
+from .. import configure_dbsession
+
+
+logger = structlog.getLogger(__name__)
+EPILOG = __doc__
+
+
+def purge_item_type_from_storage(app, item_types, prod=False):
+    """
+    Purges all items with the given item_types from our storage. This function could partially purge
+    an item type if an error is encountered. Note that this will work no matter what resources are backing
+    'PickStorage'.
+
+    IMPORTANT: If an error occurs the DB transaction is rolled back, but the ES deletions will persist.
+               Re-running 'create_mapping' on this item type will reindex the items.
+
+    :param app: app to access settings from, either a testapp (testapp.app.registry) or regular app(.registry)
+    :param item_types: list of types to purge from DB
+    :param prod: bool whether to allow run on prod, default False
+    :return: True in success, False otherwise
+    """
+    if not hasattr(app, 'registry'):
+        app = app.app
```

Comment:
What is this about? I don't see other endpoints doing this. Is this some weird heuristic fix that tries to assume someone passed the module for `app.py` rather than the value of `app` in that module?? That's my best guess. :)

Reply:
Sometimes there is another level of indirection. I'd like to be able to tolerate both.